pkg/chunked/storage_linux.go

```diff
@@ -1487,7 +1487,7 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 	for i := range mergedEntries {
 		r := &mergedEntries[i]
 		if options.ForceMask != nil {
-			value := idtools.FormatContainersOverrideXattr(r.UID, r.GID, int(r.Mode))
+			value := idtools.FormatContainersOverrideXattrDevice(r.UID, r.GID, int(r.Mode), int(r.Devmajor), int(r.Devminor))
```

Comment:
Doesn’t this need to handle `r.Type`?

And there is a `SetContainersOverrideXattr` call in this file that seems to need updating.